grammar/rainerscript.c

```diff
@@ -2374,66 +2447,16 @@ doFuncCall(struct cnffunc *__restrict__ const func, struct svar *__restrict__ co
 		cnfexprEval(func->expr[0], &r[0], usrptr, pWti);
 		str = (char*) var2CString(&r[0], &bMustFree);
 
+		bMustFree2 = 0;
+
 		// Check if the optional 2nd parameter was provided
 		if(func->nParams == 2) {
 			cnfexprEval(func->expr[1], &r[1], usrptr, pWti);
 			fmt = (char*) var2CString(&r[1], &bMustFree2);
 		}
 
 		ret->datatype = 'N';
-		ret->d.n = 0;
-		wtiSetScriptErrno(pWti, RS_SCRIPT_EOK);
-
-		if (objUse(datetime, CORE_COMPONENT) == RS_RET_OK) {
-			struct syslogTime s;
-			int len = strlen(str);
-			uchar *pszTS = (uchar*) str;
-
-			int numFormats  = 3;
-			const char *formats[] = { "date-rfc3339", "date-rfc3164", "date-unix" };
-			char **pf = (char *[]) { NULL };
-			char **p  = (char **) formats;
-
-			// Check if a format specifier was explicitly provided
-			if (fmt != NULL) {
-				numFormats = 1;
-				*pf = fmt;
-				p = pf;
-			}
-
-			// Enumerate format specifier options, looking for the first match
-			for (int i = 0; i < numFormats; i++) {
-				char *f = p[i];
-
-				if (strcmp(f, "date-rfc3339") == 0) {
-					if (datetime.ParseTIMESTAMP3339(&s, (uchar**) &pszTS, &len) == RS_RET_OK) {
-						DBGPRINTF("is_time: RFC3339 format found.\n");
-						ret->d.n = 1;
-						break;
-					}
-				} else if (strcmp(f, "date-rfc3164") == 0) {
-					if (datetime.ParseTIMESTAMP3164(&s, (uchar**) &pszTS, &len,
-						NO_PARSE3164_TZSTRING, NO_PERMIT_YEAR_AFTER_TIME) == RS_RET_OK) {
-						DBGPRINTF("is_time: RFC3164 format found.\n");
-						ret->d.n = 1;
-						break;
-					}
-				} else if (strcmp(f, "date-unix") == 0) {
-					var2Number(&r[0], &retval);
-					if (retval) {
-						DBGPRINTF("is_time: UNIX format found.\n");
-						ret->d.n = 1;
-						break;
-					}
-				} else {
-					DBGPRINTF("is_time: %s is not a valid date/time format specifier!\n", f);
-					break;
-				}
-			}
-		}
-
-		// If not a valid date/time string, set 'errno'
-		if (ret->d.n == 0) wtiSetScriptErrno(pWti, RS_SCRIPT_EINVAL);
+		ret->d.n = doFunc_is_time(str, fmt, &r[0], pWti);
 
 		if(bMustFree) free(str);
 		if(bMustFree2) free(fmt);
```

Comment:
... which means both of the lines can be removed -- or am I overlooking something?